pjsip/src/pjsua-lib/pjsua_acc.c

```diff
@@ -3425,13 +3425,22 @@ pj_status_t pjsua_acc_get_uac_addr(pjsua_acc_id acc_id,
     }
 
     /* Get transport type of the URI */
-    if (PJSIP_URI_SCHEME_IS_SIPS(sip_uri))
-        tp_type = PJSIP_TRANSPORT_TLS;
-    else if (sip_uri->transport_param.slen == 0) {
+    if (sip_uri->transport_param.slen == 0) {
         tp_type = PJSIP_TRANSPORT_UDP;
     } else
         tp_type = pjsip_transport_get_type_from_name(&sip_uri->transport_param);
 
+    /* If the URI uses sips scheme, make sure we use secure transport. */
+    if (PJSIP_URI_SCHEME_IS_SIPS(sip_uri)) {
+        unsigned flag, tp_flag;
+
+        tp_flag = (PJSIP_TRANSPORT_SECURE | PJSIP_TRANSPORT_RELIABLE);
```

Comment:
why `RELIABLE` is also added?

Reply:
It's the same requirement as in `pjsip_get_dest_info()`:
https://github.com/pjsip/pjproject/blob/master/pjsip/src/pjsip/sip_util.c#L829

The code doesn't provide the reason and I can't find any recommendation in the RFC, but I believe this is because in practice, most secure transport protocols are also designed to be reliable, because the security mechanisms are useless if the data is not delivered successfully.

Reply:
Was just wondering if reliability is required too by SIPS by standard, because I am not sure if websocket is reliable (so if someone implemented it in the future, there will be `RELIABLE` flag in it, or perhaps whether SIPS using DTLS is allowed). Or perhaps it is because currently we only has TLS for SIP secure transport which is reliable?

Reply:
So do you think we should remove the reliability requirement or just leave it as is?

Reply:
Well, that's my question actually :)

From a quick search, it looks like SIPS over DTLS was considered:
https://www.ietf.org/archive/id/draft-jennings-sip-dtls-05.html
```
Requests with a SIPS URI can be sent over DTLS as well as TLS. 
```
The draft never becomes RFC though.

Anyway, from the draft above and other SIPs related RFCs which do not seem to explicitly mention about reliable/unreliable transports, it looks like reliability is not mandatory in SIPS, so IMO it should be removed. But not sure about the practicality, especially for PJSIP which currently only has TLS.

Reply:
Okay, I think I found out the reason why RELIABLE flag was included. I believe it's because the function `pjsip_transport_get_type_from_flag()` now searches for exact match for the specified flags. So I modified it to find transport that contains the flags instead.

Note that this will cause a behavioural change. Within the library itself, the function is currently only used in `pjsip_get_dest_info()` in `sip_util.c`, so there's no issue. While for apps that directly call the function, it should have no problem as well, except perhaps for apps that significantly modify SIP transports.